### ISSUE TITLE
Onfido should not require camera, microphone features

### ISSUE THREAD

**Issue by amit-affirm** [onfido/onfido-android-sdk#136]
Currently, because we're integrating Onfido sdk, our app is getting marked by Google Play that it requires Camera and Microphone features. That is not the case, and we would like Onfido to mark these features as being optional features.

The below is one snippet you can use for example.

```
<uses-feature android:name="android.hardware.camera" android:required="false"/>
<uses-feature android:name="android.hardware.microphone" android:required="false"/>
```

It is safe to do this as long as Onfido sdk fails gracefully when this hardware is not present, which it currently does.

#### What was the expected behaviour?
Expected that our app doesn't require any new features.

#### What happened instead?
Google play reported that we have started requiring some new features so app will no longer be available to some devices.

#### Version info:

- `onfido-capture-sdk` version: onfido-capture-sdk-core:7.3.0
- Android API version(s): 29

- `compileSdkVersion`: 29
- `targetSdkVersion`: 29
- `minSdkVersion`: 23
- Support library version (if applicable):

#### Steps to reproduce:
You can run `aapt dump badging <path_to_exported_.apk>` on the sample app

#### Other useful info:
_screenshots, videos, etc._

**Comment by burakeregar** [comment]
Hi @amit-affirm 

>It is safe to do this as long as Onfido sdk fails gracefully when this hardware is not present, which it currently does.
- Yes, Onfido SDK fails this by showing a related message to the user. However, if Onfido's verification flow is mandatory for your app, end-users won't be able to pass Onfido whatsoever. So, why would you prefer this, over overriding required parameter and checking [whether the device has a camera](https://stackoverflow.com/questions/1944117/check-if-device-has-a-camera) even before launching the Onfido flow to save some time for the end user? 

If the hardware is mandatory for software and it doesn't state that in the Manifest, that may cause unexpected behaviours to some integrators which are not even aware of it. In this case, if you are overriding a mandatory hardware requirement, you know that you need to handle this.

**Comment by amit-affirm** [comment]
> If the hardware is mandatory for software and it doesn't state that in the Manifest, that may cause unexpected behaviours to some integrators which are not even aware of it. In this case, if you are overriding a mandatory hardware requirement, you know that you need to handle this.

Yes, I agree. The question is, is camera mandatory for Onfido sdk? If yes, then it should mark it as _required_. If not, then it should mark it is as _not required_ in manifest. Integrator should not have to override this.

**Comment by burakeregar** [comment]
Yes, camera is mandatory for Onfido SDK. SDK displays a message to the user if the device doesn't have a camera, however, it doesn't necessarily mean that it's been handled properly. For example, as an integrator, you won't be aware of this case and you will see a drop-off in the Onfido flow. The ideal scenario would depend on the integrator's use-case to handle this accordingly and ideally being handled even before launching the identity verification flow.

For example, for the microphone hardware requirement case, it's not stated as required in the manifest as we don't record any audio for the doc and selfie flows.

**Comment by amit-affirm** [comment]
> Yes, camera is mandatory for Onfido SDK. SDK displays a message to the user if the device doesn't have a camera, however, it doesn't necessarily mean that it's been handled properly. For example, as an integrator, you won't be aware of this case and you will see a drop-off in the Onfido flow. The ideal scenario would depend on the integrator's use-case to handle this accordingly and ideally being handled even before launching the identity verification flow.

OK thanks.


> For example, for the microphone hardware requirement case, it's not stated as required in the manifest as we don't record any audio for the doc and selfie flows.

You can see microphone feature is coming up by running `aapt dump badging app-universal-release-unsigned.apk` on the sample app. Seems like a bug.

```
 uses-feature: name='android.hardware.camera'
  uses-feature-not-required: name='android.hardware.camera.autofocus'
  uses-feature: name='android.hardware.faketouch'
  uses-implied-feature: name='android.hardware.faketouch' reason='default feature for all apps'
  **uses-feature: name='android.hardware.microphone'**
  uses-implied-feature: name='android.hardware.microphone' reason='requested android.permission.RECORD_AUDIO permission'
  uses-feature: name='android.hardware.screen.portrait'
  uses-implied-feature: name='android.hardware.screen.portrait' reason='one or more activities have specified a portrait orientation'
```

**Comment by burakeregar** [comment]
It seems like the reason is RECORD_AUDIO permission. I have created an internal ticket to investigate this issue. We'll update here once we have any update.
